server/src/test/java/io/crate/analyze/expressions/IntervalAnalysisTest.java

```diff
@@ -44,37 +44,65 @@ public void prepare() throws Exception {
+    public void test_psql_compact_format_from_string_with_start_year2() {
+        var symbol = e.asSymbol("'6 years 5 mons 4 days 03:02:01'::interval");
+        assertThat(symbol).isLiteral(new Period().withYears(6)
+            .withMonths(5).withDays(4).withHours(3).withMinutes(2).withSeconds(1)
+            .withPeriodType(PeriodType.yearMonthDayTime()));
     }
 
     @Test
     public void test_psql_compact_format_from_string_with_start_end() {
         var symbol = e.asSymbol("INTERVAL '6 years 5 mons 4 days 03:02:01' YEAR TO MONTH");
         assertThat(symbol).isLiteral(new Period().withYears(6).withMonths(5)
-                                         .withPeriodType(PeriodType.yearMonthDayTime()));
+            .withDays(0).withHours(0).withMinutes(0)
+            .withPeriodType(PeriodType.yearMonthDayTime()));
     }
 
     @Test
     public void test_psql_compact_format_from_string_with_start_end1() {
         var symbol = e.asSymbol("INTERVAL '6 years 5 mons 4 days 03:02:01' DAY TO HOUR");
         assertThat(symbol).isLiteral(new Period().withYears(6).withMonths(5).withDays(4).withHours(3)
-                                         .withPeriodType(PeriodType.yearMonthDayTime()));
+            .withMinutes(0).withSeconds(0)
+            .withPeriodType(PeriodType.yearMonthDayTime()));
+    }
+
+    @Test
+    public void test_psql_compact_format_from_string_with_start_end_day_minute() {
+        var symbol = e.asSymbol("INTERVAL '6 years 5 mons 4 days 03:02:01' DAY TO MINUTE");
+        assertThat(symbol).isLiteral(new Period().withYears(6).withMonths(5).withDays(4).withHours(3).withMinutes(2)
+            .withSeconds(0)
+            .withPeriodType(PeriodType.yearMonthDayTime()));
+    }
+
+    @Test
+    public void test_psql_compact_format_from_string_with_start_end_day_seconds() {
+        var symbol = e.asSymbol("INTERVAL '6 years 5 mons 4 days 03:02:01' DAY TO SECOND");
+        assertThat(symbol).isLiteral(new Period().withYears(6).withMonths(5).withDays(4).withHours(3).withMinutes(2).withSeconds(1)
+            .withPeriodType(PeriodType.yearMonthDayTime()));
+    }
+
+    @Test
+    public void test_psql_compact_format_from_string_with_start_end_hour_minute() {
+        var symbol = e.asSymbol("INTERVAL '6 years 5 mons 4 days 03:02:01.100' HOUR TO MINUTE");
+        assertThat(symbol).isLiteral(new Period().withYears(6).withMonths(5).withDays(4).withHours(3).withMinutes(2)
+            .withSeconds(0)
+            .withPeriodType(PeriodType.yearMonthDayTime()));
+    }
+
+    @Test
+    public void test_psql_compact_format_from_string_with_start_end_hour_seconds() {
+        var symbol = e.asSymbol("INTERVAL '6 years 5 mons 4 days 03:02:01' HOUR TO SECOND");
+        assertThat(symbol).isLiteral(new Period().withYears(6).withMonths(5).withDays(4).withHours(3).withMinutes(2).withSeconds(1)
+            .withPeriodType(PeriodType.yearMonthDayTime()));
```

Comment:
This is not related to MILLISECOND change, I think we can remove it. 

On the other hand, it's better coverage of the existing functionality, isn't it? 
If you want, we can keep it but let's put those changes to a separate commit on order not to mix it with MILLISECONDS unit changes.

Reply:
Thanks. I will make a separate commit for these tests.